lib/collection/src/collection/mod.rs

```diff
@@ -283,6 +287,7 @@ impl Collection {
             update_runtime: update_runtime.unwrap_or_else(Handle::current),
             search_runtime: search_runtime.unwrap_or_else(Handle::current),
             optimizer_cpu_budget,
+            local_stats_cache: LocalDataStatsCache::default(),
```

Comment:
What about computing the stats once on collection load as well?
Otherwise it seems the 32 first requests will not benefit from the cache?
WDYT?

Reply:
The cache gets loaded on the first request since [this](https://github.com/qdrant/qdrant/pull/5501/files#diff-f91d53257ec619a1403e6122ec4c7f004fffe1f32bd91286f6950dbcab39c3a5R20) check evaluates to `true` when called the first time.
We however can still load the stats after loading the collection to improve on performance 👍🏻